server/text_generation_server/layers/attention/kv_cache.py

```diff
@@ -24,11 +24,13 @@ def __init__(
     ):
         """Construct the key-value cache for a layer."""
 
-        if dtype == torch.float8_e5m2 and (
-            ATTENTION != "flashinfer" or SYSTEM != "cuda"
+        if (
+            dtype.itemsize == 1
+            and dtype.is_floating_point
```

Comment:
Why not `in {torch.float8_e5m2, torch.float8_e4m3}` ?

Reply:
We might have other variants in the future as well (`e4m3fnuz`, `e5m2uz`).

Reply:
Fair enough. From my understanding they are actually the same dtypes only side effectifs on edge values (which we won't check against).

I don't even think torch have plans for explicit support https://pytorch.org/docs/stable/tensors.html#data-types

As you wish for merging, I just feel like having consistent checks everywhere will lead to fewer bugs.

Reply:
Yeah, that's a fair point. I think `in {torch.float8_e5m2, torch.float8_e4m3}` makes it easier to grep for all the places where one of these float types is used. So I'll update the PR.